driver/main.c

```diff
@@ -1231,15 +1404,15 @@ static long ppm_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 	case PPM_IOCTL_DISABLE_SIGNAL_DELIVER:
 	{
 		vpr_info("PPM_IOCTL_DISABLE_SIGNAL_DELIVER\n");
-		if (g_tracepoint_registered)
+		if (g_tracepoints_attached & (1 << SIGNAL_DELIVER))
 			compat_unregister_trace(signal_deliver_probe, "signal_deliver", tp_signal_deliver);
 		ret = 0;
 		goto cleanup_ioctl;
 	}
 	case PPM_IOCTL_ENABLE_SIGNAL_DELIVER:
 	{
 		vpr_info("PPM_IOCTL_ENABLE_SIGNAL_DELIVER\n");
-		if (g_tracepoint_registered)
+		if (!(g_tracepoints_attached & (1 << SIGNAL_DELIVER)))
 			compat_register_trace(signal_deliver_probe, "signal_deliver", tp_signal_deliver);
 		ret = 0;
 		goto cleanup_ioctl;
```

Comment:
So now we will have two ways for doing this, same for the page faults, do we want to keep them for backward compatibility?

Reply:
I am not really sure; wasn't too keen on breaking stuff :) 
@gnosek  wdyt?